### DISCUSSION TITLE
How to refresh init data?

### DISCUSSION THREAD

**Discussion by krutoo** [Telegram-Mini-Apps/tma.js#585] [Q&A] [answered]
Hi in my mini app i validate data in nodejs from frontend request by validate function from sdk package

I provided expiresIn parameter but i want to make frontend able to refresh authorization data and make session not limited by time (like jwt for example) 

How can i refresh init data in frontend js?

**Comment by heyqbnk** [comment, marked as answer]
Implement your own handler on your backend and return new init data. Telegram Mini Apps doesn't provide such functionality and will probably not. Each mini application run receives its own init data.

**Reply by krutoo** [reply]
@heyqbnk thank you a lot for your answer. Can you provide minimal example of this approach?

I don't fully understand what endpoint I should add to the backend and what it should accept and return

**Reply by heyqbnk** [reply]
Sorry, I have no time for that.

You can create init data using signing functions:
https://docs.telegram-mini-apps.com/packages/telegram-apps-init-data-node#signing
